src/main/java/org/truffleruby/extra/ffi/Pointer.java

```diff
@@ -143,7 +143,9 @@ public int findByte(long offset, byte value) {
     }
 
     public void free() {
-        UNSAFE.freeMemory(address);
+        if (!autorelease) {
+            UNSAFE.freeMemory(address);
+        }
```

Comment:
This is racy with enableAutorelease/disableAutorelease but not sure what we can do about it.
Synchronizing `free()` would be rather expensive.